sdk/Lusid.Sdk/Utilities/LusidExceptionHandler.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using Lusid.Sdk.Client;
+using RestSharp;
+
+namespace Lusid.Sdk.Utilities
+{
+    /// <summary>
+    /// Handles the generation of LUSID exceptions after receiving the ApiResponse
+    /// </summary>
+    public static class LusidExceptionHandler
+    {
+        /// <summary>
+        /// Generate exceptions from the ApiResponse when ResponseStatus is an Error,
+        /// and StatusCode has no content or is less than 400
+        /// </summary>
+        /// <param name="methodName">The name of the method</param>
+        /// <param name="response">The ApiResponse</param>
+        public static Exception CustomExceptionFactory(string methodName, IApiResponse response)
+        {
+            var status = (int)response.StatusCode;
+            
+            // Throw if status code is >= 400
+            if (status >= 400)
+            {
+                return new ApiException(status,
+                    $"Error calling {methodName}: {response.RawContent}",
+                    response.RawContent, response.Headers);
+            }
```

Comment:
This already exists on the default exception handler, why not just call that?

Reply:
could do to be fair. But since this is custom, I wanted to keep it separate from any default implementations. What do you think is best?